packages/client/README.md

```diff
@@ -1,232 +1,296 @@
+    <img alt="Specification" src="https://img.shields.io/static/v1?label=specification&message=v0.6.0&color=yellow&style=for-the-badge" />
+  </a>
+  <!-- x-release-please-start-version -->
+  <a href="https://github.com/open-feature/js-sdk/releases/tag/web-sdk-v0.4.0">
+    <img alt="Release" src="https://img.shields.io/static/v1?label=release&message=v0.4.0&color=blue&style=for-the-badge" />
+  </a>
+  <!-- x-release-please-end -->
+  <br/>
+  <a href="https://www.repostatus.org/#wip">
+    <img alt="Project Status" src="https://www.repostatus.org/badges/latest/wip.svg" />
+  </a>
+  <a href="https://open-feature.github.io/js-sdk/modules/OpenFeature_Web_SDK.html">
+    <img alt="API Reference" src="https://img.shields.io/badge/reference-teal?logo=javascript&logoColor=white" />
+  </a>
+  <a href="https://www.npmjs.com/package/@openfeature/web-sdk">
+    <img alt="NPM Download" src="https://img.shields.io/npm/dm/%40openfeature%2Fweb-sdk" />
+  </a>
+  <a href="https://codecov.io/gh/open-feature/js-sdk">
+    <img alt="codecov" src="https://codecov.io/gh/open-feature/js-sdk/branch/main/graph/badge.svg?token=3DC5XOEHMY" />
+  </a>
+    <a href="https://bestpractices.coreinfrastructure.org/projects/6594">
+    <img alt="CII Best Practices" src="https://bestpractices.coreinfrastructure.org/projects/6594/badge" />
+  </a>
+</p>
+<!-- x-hide-in-docs-start -->
 
-[OpenFeature][openfeature-website] is an open standard that provides a vendor-agnostic, community-driven API for feature
-flagging that works with your favorite feature flag management tool.
+[OpenFeature](https://openfeature.dev) is an open standard that provides a vendor-agnostic, community-driven API for feature flagging that works with your favorite feature flag management tool.
 
-### Why standardize feature flags?
+<!-- x-hide-in-docs-end -->
 
-Standardizing feature flags unifies tools and vendors behind a common interface which avoids vendor lock-in at the code
-level. Additionally, it offers a framework for building extensions and integrations and allows providers to focus on
-their unique value proposition.
+## 🚀 Quick start
 
-## 🔍 Requirements:
+### Requirements
 
 - ES2015-compatible web browser (Chrome, Edge, Firefox, etc)
 
-## 📦 Installation:
+### Install
 
-### npm
+#### npm
 
 ```sh
-npm install @openfeature/web-sdk
+npm install --save @openfeature/web-sdk
 ```
 
-### yarn
+#### yarn
 
 ```sh
 yarn add @openfeature/web-sdk
 ```
 
-## 🌟 Features:
+### Usage
 
-- support for various [providers](https://openfeature.dev/docs/reference/concepts/provider)
-- easy integration and extension via [hooks](https://openfeature.dev/docs/reference/concepts/hooks)
-- handle flags of any type: bool, string, numeric and object
-- [context-aware](https://openfeature.dev/docs/reference/concepts/evaluation-context) evaluation
+```ts
+import { OpenFeature } from '@openfeature/web-sdk';
 
-## 🚀 Usage:
+// Register your feature flag provider
+OpenFeature.setProvider(new YourProviderOfChoice());
```

Comment:
The current readme shows that `setProvider` returns a promise. However, that doesn't appear to be the case.

I would like this basic example to wait for initialization before performing flag evaluation. What's the recommended approach here? Is it possible to set context before initializing a new provider?

Reply:
> The current readme shows that setProvider returns a promise. However, that doesn't appear to be the case.

Yes, this not the case anymore. For both SDK, you get back the Api without waiting for initialization.

> Is it possible to set context before initializing a new provider?

Yes this can be set at any point. When `initialize` on the provider is called, the current context is given into it.

> I would like this basic example to wait for initialization before performing flag evaluation. What's the recommended approach here?

I think this would only be possible with a top level await, where a promise is resolved wen the ready handler is called. Without this, you can not really wait for the initialization.